gen/templates/component/component-name.adb

```diff
@@ -943,12 +943,31 @@ package body Component.{{ name }} is
 
    not overriding procedure Process_Parameter_Update (Self : in out Base_Instance; Par_Update : in out Parameter_Update.T) is
       use Parameter_Operation_Type;
+      use Parameter_Validation_Status;
+      use Parameter_Update_Status;
       Status : Parameter_Update_Status.E := Parameter_Update_Status.Success;
    begin
       case Par_Update.Operation is
          when Stage =>
             -- Stage this parameter.
             Status := Self.Stage_Parameter (Par_Update.Param);
+         when Validate =>
+            -- Pass the staged parameters to the user defined validation function.
+            -- Note that type ranges have already been validated as part of staging
+            -- each parameter individually. This function is used for extended,
+            -- user-implemented validation.
+            if Base_Instance'Class (Self).Validate_Parameters (
+{% for par in parameters %}
+               {{ par.name }} => Self.Staged_Parameters.Get_{{ par.name }}{{ "," if not loop.last }}
+{% endfor %}
+            ) /= Valid
+            then
+               -- Set status:
+               Status := Validation_Error;
+            else
+               -- Set status:
+               Status := Success;
+            end if;
```

Comment:
This would be much clearer as a case statement, since it will show all the possible returns.

Reply:
Agreed! Addressed in [1b827a7](https://github.com/lasp/adamant/pull/64/commits/1b827a780e24f58930e9c35e67bc685087cccec5).